transaction/_transaction.py

```diff
@@ -760,3 +753,13 @@ def __init__(self, datamanager):
 
     def rollback(self):
         raise TypeError("Savepoints unsupported", self.datamanager)
+
+def text_or_warn(s):
+    if isinstance(s, text_type):
+        return s
+
+    warnings.warn("Expected text", DeprecationWarning, stacklevel=3)
+    if isinstance(s, bytes):
+        return s.decode('utf-8', 'replace')
+    else:
+        return text_type(s)
```

Comment:
FWIW, RelStorage 1.6 simply ran `str(txn.user)` and `str(txn.description)`, so it could take anything for those values.